### ISSUE TITLE
Add Liquid Lexer

### ISSUE THREAD

**Issue by parkr** [jneen/rouge#185]
Hi! I would love to see a lexer for Liquid. One is in pygments.

- `LiquidLexer` class in Pygments: https://bitbucket.org/birkenfeld/pygments-main/src/53092018461a95c3310adc276f1ef7f6d7715b50/pygments/lexers/templates.py#cl-1857
- Changeset: https://bitbucket.org/birkenfeld/pygments-main/commits/53092018461a
- Original issue: https://bitbucket.org/birkenfeld/pygments-main/issue/977/add-liquid-lexer
- Pygments lexer by @maul-esel: https://gist.github.com/maul-esel/6652045

Not sure how to go about converting this to Rouge but wanted to open the issue so we can track it.

**Comment by RadLikeWhoa** [comment]
I started working on this today and it's mostly finished, there's just some small bugs that still need to be fixed. I little to no experience in Ruby so I ended up porting the Pygments lexer, which worked just fine, but now I'm stuck.

Here's what I have so far: https://github.com/RadLikeWhoa/rouge/blob/liquid-lexer/lib/rouge/lexers/liquid.rb

My problem is that I want to pass a RegEx group to another state, but I don't know how. For example, on [line 20](https://github.com/RadLikeWhoa/rouge/blob/liquid-lexer/lib/rouge/lexers/liquid.rb#L20) I want to pass the third group (i.e. replacing `Name::Variable`) through the `:generic` state. Is this possible? (I hope my question is clear enough. :))

Most of the remaining work hangs on this problem so any help would be appreciated.
